render_hawk_2/camera_class.cs

```diff
@@ -0,0 +1,71 @@
+class camera_class
+{
+    public double[] position;
+    public double[] vector_look;
+    public double[] vector_up;
+    public double[] vector_u;
+    public double[] vector_v;
+    public double view_angle_width;
+    public double view_angle_height;
+    private System.Random random;
+
+    private double[] output_vector;
+    private double x, y, angle_x, angle_y, image_width_half, image_height_half, u_scalar, v_scalar, w_scalar, pi_180_half, cos_angle_y;
+
+    public camera_class ()
+    {
+        position = new double[3];
+        vector_look = new double[3];
+        vector_up = new double[3];
+        output_vector = new double[3];
+        view_angle_width = 0;
+        view_angle_height = 0;
+        random = new System.Random ();
+
+        pi_180_half = System.Math.PI / 180 * 0.5;
+    }
+
+    public double[] get_vector (scene_class scene, int x_pixel, int y_pixel)

```

Comment:
No need to pass in the entire scene here, just pass the information necessary (the camera).